.github/workflows/precommit.yml

```diff
@@ -49,3 +49,27 @@ jobs:
         continue-on-error: true
         with:
           pylance-version: latest-release
+
+  js-pre-commit:
+    name: JS Pre-commit checks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "20"
+
+      - name: Install dependencies
+        working-directory: ./javascript-sdk
```

Comment:
we should just call the folder `js`. it's not SDK, and JS a pretty widespread short name